src/impls.rs

```diff
@@ -58,6 +59,51 @@ where
     }
 }
 
+impl<T> Diff for Box<T>
+where
+    T: Diff + Clone,
+{
+    type Repr = T::Repr;
```

Comment:
I think we should wrap the diff'ed type into a Box as well, like `type Repr = Box<T::Repr>`. Boxes let users create arbitrarily sized nested data structures, like singly linked lists. If the diff mimics their format but ignores the boxes, it'll generate an invalid struct. For example, this will cause a compiler error in this state:

```rust
#[derive(Debug, PartialEq, Diff)]
struct LinkedListNode<T: Diff + PartialEq> {
    value: T,
    child: Option<Box<LinkedListNode<T>>>,
}
```